File: src/main/java/com/github/johanfredin/bindygenerator/FieldType.java
package com.github.johanfredin.bindygenerator;

public class FieldType implements Comparable<FieldType>{

    private int priority;
    private String type;

    public FieldType(int priority, String type) {
        this.priority = priority;
        this.type = type;
    }

    public int getPriority() {
        return priority;
    }

    public void setPriority(int priority) {
        this.priority = priority;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    @Override
    public int compareTo(FieldType fieldType) {
        if(this.priority > fieldType.priority) {
            return 1;
        } else if(this.priority < fieldType.priority) {
            return -1;
        }
        return 0;
    }
}

File: src/main/java/com/github/johanfredin/bindygenerator/FieldType.kt
package com.github.johanfredin.bindygenerator

/**
 * Used when wanting numeric fields in resulting java class.
 * Holds a objectTypeName and the priority of the objectTypeName. E.g if field in
 * datasource is numeric on one line and a string on the next
 * then String will be used for the variable etc.
 * @property priority the priority of the variable objectTypeName
 */
enum class FieldType(val priority: Int) : Comparable<FieldType> {

    STRING(1),
    DECIMAL(2),
    INTEGER(3);

}
